NoteFind/View Controllers/SearchViewController.m

```diff
@@ -23,86 +23,45 @@ - (void)viewDidLoad {
-    self.isFiltered = NO;
-    self.allTagTitles = [[NSMutableArray alloc] init];
-
-    self.allTags = [[NSMutableArray alloc] init];
-    [self getTags];
-    
-    
-    
 }
 
-- (void) getTags {
-    PFQuery *tagQuery = [Tags query];
-    [tagQuery orderByAscending:@"title"];
-    [tagQuery includeKey:@"title"];
-    NSMutableArray *tagTitles = [[NSMutableArray alloc]init];
-    
-    [tagQuery findObjectsInBackgroundWithBlock:^(NSArray<Tags *> *tags, NSError *error) {
-        if (tags) {
-            // adding title to allTags for table view
-            for (Tags *tag in tags) {
-                NSString *title = tag.title;
-                [tagTitles addObject:title];
-
-                [self.allTags addObject:tag];
-            }
-            self.allTagTitles = (NSArray *)tagTitles;
-
-           
-        } else {
-            // TODO: handle error
-        }
-        [self.tableView reloadData];
-    }];
-}
 
 - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
     UITableViewCell *cell = [self.tableView dequeueReusableCellWithIdentifier:@"Cell" forIndexPath:indexPath];
         
-    if (self.isFiltered) {
-
-        cell.textLabel.text = self.displayTagTitles[indexPath.row];
-    }
-    
-    else {
-        cell.textLabel.text = self.allTagTitles[indexPath.row];
-
-
-    }
+    cell.textLabel.text = self.displayTagTitles[indexPath.row];
     
     return cell;
     
 }
 
 - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
-    if (self.isFiltered) {
-
-        return self.displayTagTitles.count;
-    }
     
-    return self.allTagTitles.count;
-
+    return self.displayTagTitles.count;
 }
 
 - (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
-    if (searchText.length == 0) {
-        self.isFiltered = NO;
-    } else {
-        self.isFiltered = YES;
-
-        self.displayTagTitles = [[NSMutableArray alloc]init];
-        for (NSString *tag in self.allTagTitles) {
-            NSRange range = [tag rangeOfString:searchText options:NSCaseInsensitiveSearch];
-            if (range.location != NSNotFound) {
-                [self.displayTagTitles addObject:tag];
-
+    //removing white space
+    searchText = [searchText stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
+    PFQuery *tagQuery = [Tags query];
+    [tagQuery whereKey:@"title" matchesRegex:searchText modifiers:@"i"];
+    [tagQuery orderByAscending:@"title"];
+    self.displayTagTitles = [[NSMutableArray alloc] init];
+    self.allTags = [[NSMutableArray alloc] init];
+    [tagQuery findObjectsInBackgroundWithBlock:^(NSArray<Tags *>*tags , NSError *error) {
+        if (tags) {
+            for (Tags *tag in tags) {
+                if (![self.displayTagTitles containsObject:tag.title]) {
+                    [self.displayTagTitles addObject:tag.title];
+                    [self.allTags addObject:tag];
```

Comment:
What is `allTags` used for now?

Reply:
allTags store the actual tag objects. They'll have the same index as their associated title and are used in the segue so we know which tag to pass into the SearchDetailsViewController

Reply:
Fixed - combined them into one mutable array